File: clients/javascript/src/apis/msgsNamespace.ts
// this file is @generated

import {
    type MsgNamespaceCreateIn,
    MsgNamespaceCreateInSerializer,
} from '../models/msgNamespaceCreateIn';
import {
    type MsgNamespaceCreateOut,
    MsgNamespaceCreateOutSerializer,
} from '../models/msgNamespaceCreateOut';
import {
    type MsgNamespaceGetIn,
    MsgNamespaceGetInSerializer,
} from '../models/msgNamespaceGetIn';
import {
    type MsgNamespaceGetOut,
    MsgNamespaceGetOutSerializer,
} from '../models/msgNamespaceGetOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class MsgsNamespace {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Creates or updates a msgs namespace with the given name. */
    public create(
        msgNamespaceCreateIn: MsgNamespaceCreateIn,
        ): Promise<MsgNamespaceCreateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/msgs/namespace/create");

        request.setBody(
            MsgNamespaceCreateInSerializer._toJsonObject(
                msgNamespaceCreateIn,
            )
        );
        return request.send(
            this.requestCtx,
            MsgNamespaceCreateOutSerializer._fromJsonObject,
        );
    }/** Gets a msgs namespace by name. */
    public get(
        msgNamespaceGetIn: MsgNamespaceGetIn,
        ): Promise<MsgNamespaceGetOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1/msgs/namespace/get");

        request.setBody(
            MsgNamespaceGetInSerializer._toJsonObject(
                msgNamespaceGetIn,
            )
        );
        return request.send(
            this.requestCtx,
            MsgNamespaceGetOutSerializer._fromJsonObject,
        );
    }
}

File: clients/rust/src/api/msgs_namespace.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct MsgsNamespace<'a> {
    cfg: &'a Configuration,
}

impl<'a> MsgsNamespace<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Creates or updates a msgs namespace with the given name.
    pub async fn create(
        &self,
        msg_namespace_create_in: MsgNamespaceCreateIn,
    ) -> Result<MsgNamespaceCreateOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/msgs/namespace/create")
            .with_body(msg_namespace_create_in)
            .execute(self.cfg)
            .await
    }

    /// Gets a msgs namespace by name.
    pub async fn get(&self, msg_namespace_get_in: MsgNamespaceGetIn) -> Result<MsgNamespaceGetOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/msgs/namespace/get")
            .with_body(msg_namespace_get_in)
            .execute(self.cfg)
            .await
    }
}
